synapse/handlers/message.py

```diff
@@ -1538,6 +1545,15 @@ async def _persist_events(
                     # stream_ordering entry manually (as it was persisted on
                     # another worker).
                     event.internal_metadata.stream_ordering = stream_id
+
+                if ratelimit:
+                    is_admin_redaction = await self.is_admin_redaction(
+                        event.type, event.sender, event.redacts
+                    )
+                    await self.request_ratelimiter.ratelimit(
+                        requester, is_admin_redaction=is_admin_redaction
+                    )
```

Comment:
This is added here to ensure we keep the ratelimits in sync with the other worker?

Reply:
Hmm, actually we probably want to:

1. Always call this, including when we get a 429 from persister
2. We should ensure this new ratelimiter doesn't actually ratelimit if the perister has persisted the event

:thinking:

Reply:
I wonder if instead we should just move this *before* us tying to persist the event.

Reply:
> I wonder if instead we should just move this before us tying to persist the event.

That sounds a lot simpler to me.